### ISSUE TITLE
[CLI] Automatic generation of CRUD operations 

### ISSUE THREAD

**Issue by 19majkel94** [19majkel94/type-graphql#66]
As TypeORM can generate sql tables from classes, it would be possible to generate resolver based on object type class data from decorators.

This way we can take TypeORM classes, mark with TypeGraphQL decorators and generate the resolvers for CRUD methods.

**Comment by stonecold123** [comment]
Will it generate typescript definition files like graphql-code-generator or apollo-codegen?

**Comment by 19majkel94** [comment]
Typescript definition files of what? All the schema is defined by code, for all your client queries you can use `apollo-codegen`.

**Comment by 19majkel94** [comment]
Thanks for the research about tools for creating the CLI 😉 I will definitely take a look at them when I start working on this, for now I have to think and collect CLI feature ideas. 

There is gitter for questions and discussions not related to issues, mostly due to nice github integration:
https://gitter.im/type-graphql

**Comment by omatrot** [comment]
Another idea : add the ability to export the graphql schema from classes.

**Comment by 19majkel94** [comment]
Right, it might be an option. It's just calling `printSchema` from `graphql` after `buildSchema` from `type-graphql` but it would impose specific folder/files project structure.

**Comment by kn0ll** [comment]
i saw similar comments in https://github.com/19majkel94/type-graphql/issues/81- regarding `printSchema`.

i think a decent solution might be to expose the `printSchema` example you provided as a command line tool-- except allow it to accept a module path and output locations as arguments (and that module path would resolve to a schema). something like (pseudocode... not thinking too hard...)

```
// createSchema.ts (created by user)
export function createSchema() {
  return buildSchema({
    // ...
  }
}

// generate.ts (exposed as type-graphql command line tool)
const createSchema = require(process.argv[0]);

(async () => {
  const schema = await createSchema();
  const sdl = printSchema(schema);
  // or stdout...
  await fs.writeFile(process.argv[1], sdl);
})();
```

**Comment by 19majkel94** [comment]
There are two ways of how to make CLI works:
- typegraphql.config.json (like ormconfig.json) where users set the `resolvers` and other options and could be automatically implicitly use by CLI and `buildSchema`
- your solution, where there's need to conform the interface like a module file with default export of function that return a promise of `GraphQLSchema`
They both have pros and cons so it might be better to implement them all, so users could choose which options fit them best 😉 

Thanks for your suggestion, I will take it into account in the future when the base CLI core will be ready - I will provide some docs how to add custom plugins/commands in own project or as a PR to the main repository.

**Comment by ematipico** [comment]
Are there any updates about this? I would like to give some help, maybe starting proposing a prototype

**Comment by fullofcaffeine** [comment]
+1 interested in the crud mainly for the CRUD boilerplate

**Comment by sebasalines** [comment]
I am not very experienced with building CLIs but I'd be happy to help any way I can

**Comment by ERICCHRISTOPHER12** [comment]
Dear Friend,

I need your assistance to move some funds($35 million dollar) from my bank to you for investment project, i work with Hang seng bank Hong Kong Please reply for more details  of this business deal,contact me on  

dreric332@gmail.com
Regards,
DR ERIC
